federation/cmd/federation-apiserver/app/core.go

```diff
@@ -27,14 +27,18 @@ import (
 	"k8s.io/kubernetes/federation/apis/core/v1"
 	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/rest"
+	namespaceetcd "k8s.io/kubernetes/pkg/registry/namespace/etcd"
 	serviceetcd "k8s.io/kubernetes/pkg/registry/service/etcd"
 )
 
 func installCoreAPIs(s *genericoptions.ServerRunOptions, g *genericapiserver.GenericAPIServer, f genericapiserver.StorageFactory) {
 	serviceStore, serviceStatusStorage := serviceetcd.NewREST(createRESTOptionsOrDie(s, g, f, api.Resource("service")))
+	namespaceStorage, namespaceStatusStorage, _ := namespaceetcd.NewREST(createRESTOptionsOrDie(s, g, f, api.Resource("namespaces")))
 	coreResources := map[string]rest.Storage{
-		"services":        serviceStore,
-		"services/status": serviceStatusStorage,
+		"services":          serviceStore,
+		"services/status":   serviceStatusStorage,
+		"namespaces":        namespaceStorage,
```

Comment:
nit: can we make `namespaceStorage` consistent with `serviceStore` here (or vice verse).  i.e. either Store or Storage, not both.